lib/x11utils.pm

```diff
@@ -105,9 +105,16 @@ sub ensure_unlocked_desktop {
             }, 10);
             record_info('Guest disabled display', 'Might be consequence of bsc#1168979');
         }
-        if (match_has_tag('authentication-required-user-settings') || match_has_tag('authentication-required-modify-system')) {
-            type_password;
-            assert_and_click "authenticate";
+      BSC1168979: {
+            if (match_has_tag('authentication-required-user-settings') || match_has_tag('authentication-required-modify-system')) {
+                wait_still_screen(2);
+                if (!check_screen([qw(authentication-required-user-settings authentication-required-modify-system)])) {
```

Comment:
I wouldn't even check the screen again here, instead I'd do:

```
if ($counter >= 9){
    record_soft_failure "bsc#1168979 - screenbuffer not updated after screen is locked - Previous content of screen is leaked for a short period of time when waking up the display or switching back from textmode tty";
   next;
}

```

And let the main loop to take care of the popup by, if it's still shown for the next round.

Reply:
I 'm confused, as it will still fail at that line `assert_and_click "authenticate";` because the glitch disappears after a second. Unless we get rid of that part at all, but then we will break first_boot if the pop-up appears there. Then also, what if we get a plain Xserver crash, that has nothing to do with that bug ?
That eventuality of Xserver crash is caught by that line `die 'ensure_unlocked_desktop repeated too much. Check for X-server crash.' if ($counter eq 1); `
I'd rather just put a dumb "if (check_var(ARCH, s390x)); then record_soft_failure" in shutdown.pm, as done earlier.

Reply:
> I 'm confused, as it will still fail at that line `assert_and_click "authenticate";` because the glitch disappears after a second. 

This is why, if it's the first time the loop runs, it's likely going to be bsc#1168979, so even if you try to type, the SUT will actually be on a different screen. 

skipping for the next loop, would give enough time for the X server to properly draw the actual screen that is supposed to be there.

If the Popup shows up, there's code to handle it (which is exactly the piece of code that you are modifying)

> Unless we get rid of that part at all, but then we will break first_boot if the pop-up appears there. Then also, what if we get a plain Xserver crash, that has nothing to do with that bug ?

One problem at a time :), this "glitch" it's annoying as it is, but there's no fix yet.

> That eventuality of Xserver crash is caught by that line `die 'ensure_unlocked_desktop repeated too much. Check for X-server crash.' if ($counter eq 1); `

In this case, we don't need to look at the X crashing. 

> I'd rather just put a dumb "if (check_var(ARCH, s390x)); then record_soft_failure" in shutdown.pm, as done earlier.

Which is trying to solve the problem in the wrong place :/

Reply:
I don't see how we can avoid to check that screen again. In first_boot, where the popup may actually be present, we still need to type the password and click "authenticate", and if not, we need to make sure to *not* type password and clic.
Also, if we want to probe bsc#1168979, we have to catch it before it disappears, not the other way round. No ?

Reply:
I tried that loop approach, does not work after 9 loops the popup is still here but disappears right after https://openqa.suse.de/tests/4820010#step/shutdown/6. I'll try something else.